crates/forge/bin/cmd/script/build.rs

```diff
@@ -101,109 +97,86 @@ impl ScriptArgs {
+                if id.source == std::path::Path::new(&target_fname) &&
+                    contract.bytecode.as_ref().map_or(false, |b| b.object.bytes_len() > 0)
+                {
+                    if target.is_some() {
+                        eyre::bail!("Multiple contracts in the target path. Please specify the contract name with `--tc ContractName`")
                     }
-
-                    filtered
+                    target = Some(id);
                 }
-
-                // if it's the target contract, grab the info
-                if extra.no_target_name {
-                    // Match artifact source, and ignore interfaces
-                    if id.source == std::path::Path::new(&extra.target_fname) &&
-                        contract.bytecode.as_ref().map_or(false, |b| b.object.bytes_len() > 0)
-                    {
-                        if extra.matched {
-                            eyre::bail!("Multiple contracts in the target path. Please specify the contract name with `--tc ContractName`")
-                        }
-                        *extra.dependencies = unique_deps(dependencies);
-                        *extra.contract = contract.clone();
-                        extra.matched = true;
-                        extra.target_id = Some(id.clone());
-                    }
-                } else {
-                    let (path, name) = extra
-                        .target_fname
-                        .rsplit_once(':')
-                        .expect("The target specifier is malformed.");
-                    let path = std::path::Path::new(path);
-                    if path == id.source && name == id.name {
-                        *extra.dependencies = unique_deps(dependencies);
-                        *extra.contract = contract.clone();
-                        extra.matched = true;
-                        extra.target_id = Some(id.clone());
-                    }
+            } else {
+                let (path, name) =
+                    target_fname.rsplit_once(':').expect("The target specifier is malformed.");
+                let path = std::path::Path::new(path);
+                if path == id.source && name == id.name {
+                    target = Some(id);
                 }
+            }
+        }
 
-                if let Ok(tc) = ContractBytecode::from(contract).try_into() {
-                    highlevel_known_contracts.insert(id, tc);
-                }
-
-                Ok(())
-            },
-            project.root(),
-        )?;
-
-        let target = extra_info
-            .target_id
-            .ok_or_else(|| eyre::eyre!("Could not find target contract: {}", target_fname))?;
+        target.ok_or_eyre(format!("Could not find target contract: {}", target_fname))
+    }
 
-        let (new_libraries, predeploy_libraries): (Vec<_>, Vec<_>) =
-            run_dependencies.into_iter().unzip();
+    pub fn link(
+        &self,
```

Comment:
do we need &self here or can this also be a standalone function?

Reply:
wonder if we can give this a different name e.g `link_script` for clarity?

Reply:
@Evalir you mean separate it from `ScriptArgs` and rename?

Reply:
@klkvr personally don't mind either separating it or keeping it on `ScriptArgs`—just think the fn name can be more specific 😄